### ISSUE TITLE
No unused annotation warning when externing crate

### ISSUE THREAD

**Issue by iopq** [rust-lang/rust#24505]
Put this at the top of the file
```
#![feature(core)]
extern crate core;
```

Result: this compiles without warnings in the nightly
Expected: warning: unused crate `core`, consider removing `extern crate core`

**Comment by steveklabnik** [comment]
/cc @pnkfelix 

I'm not sure this is a bug, exactly...

**Comment by alexcrichton** [comment]
I'm a little confused, do you expect the statement `extern crate core` to *not* require the `core` feature? (right now it is intentional that it does).

**Comment by iopq** [comment]
Because it is not required for my program to work. Maybe there should be a warning for `extern crate core` not being necessary?

**Comment by MatejLach** [comment]
I think what @iopq is saying is that when he puts `extern crate core` into his source code and then does not make use of the `core` feature in his program that the compiler should warn him something like `consider removing "extern crate core"`, something akin to `variable does not need to be mutable`, although I am not sure.

**Comment by iopq** [comment]
MatejLach: yes, that's exactly what I want

**Comment by alexcrichton** [comment]
@iopq there's [actually a lint for this today](https://github.com/rust-lang/rust/blob/master/src/librustc/lint/builtin.rs#L25-L29) although it may not be super well tested. Additionally, linking crates may have side effects such as pulling in C libraries or providing lang items which are not always easy to track (hence the allow-by-default).

For simple cases, though, I think that lint should suffice!
